selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/GitPullConflictTest.java

```diff
@@ -0,0 +1,199 @@
+
+  private static final String CHANGE_STRING_1 =
+      String.format("//first_change_%s", System.currentTimeMillis());
+
+  private static final String headConfPrefixConfMess =
+      "<<<<<<< HEAD\n" + "//second_change\n" + "=======\n" + CHANGE_STRING_1 + "\n" + ">>>>>>>";
+
+  private GitHub gitHub;
+  private GHRepository gitHubRepository;
+
+  @Inject private TestWorkspace ws;
+  @Inject private Ide ide;
+  @Inject private TestUser productUser;
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject
+  @Named("github.password")
+  private String gitHubPassword;
+
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private Menu menu;
+  @Inject private Git git;
+  @Inject private Events events;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Consoles consoles;
+  @Inject private TestSshServiceClient testSshServiceClient;
+  @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
+  @Inject private TestGitHubServiceClient gitHubClientService;
+
+  @BeforeClass
+  public void prepare() throws Exception {
+    gitHub = GitHub.connectUsingPassword(gitHubUsername, gitHubPassword);
+    gitHubRepository = gitHub.createRepository(REPO_NAME).create();
+    String commitMess = String.format("add-new-content %s ", System.currentTimeMillis());
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath = Paths.get(getClass().getResource("/projects/git-pull-conflict").getPath());
+    gitHubClientService.addContentToRepository(entryPath, commitMess, gitHubRepository);
+    ide.open(ws);
+  }
+
+  @AfterClass
+  public void deleteRepo() throws IOException {
+    gitHubRepository.delete();
+  }
+
+  @Test
+  public void pullConflictsTest() throws IOException {
+    // preconditions and import the test repo
+    String javaFileChange = "JavaCommentsTest";
+    String textFileChange = "GitPullTest.txt";
+    String changeContent_2 = "//second_change";
+
+    projectExplorer.waitProjectExplorer();
+    String repoUrl = String.format("https://github.com/%s/%s.git", gitHubUsername, REPO_NAME);
+    git.importJavaApp(repoUrl, PROJECT_NAME, MAVEN);
+
+    // change files in the test repo on GitHub
+    changeContentOnGithubSide(
+        String.format("%s/%s.java", pathToJavaFile, javaFileChange), CHANGE_STRING_1);
+    changeContentOnGithubSide(textFileChange, CHANGE_STRING_1);
+
+    // change the same files in the editor
+    changeJavaFileForTest(javaFileChange, changeContent_2);
+    changeTextFileForTest(textFileChange, changeContent_2);
+
+    // make pull and get the first conflict
+    performPull();
+    events.clickEventLogBtn();
+    events.waitExpectedMessage(firstMergeConflictMessage);
+
+    commitFiles();
+
+    // Make pull again and get second conflict
+    performPull();
+    events.clickEventLogBtn();
+    events.waitExpectedMessage(secondMergeConflictMessage);
+
```

Comment:
Why do we check 2 conflicts? 1 checking is not enough?

Reply:
by scenario we have changes in the same files locally and on GitHub, perform pull
1) conflict: `Checkout operation failed, the following files would be overwritten by merge: <...files....> Could not pull. Commit your changes before merging.`

then we make commit files, perform pull and have
2) conflict: `Could not pull because a merge conflict is detected in the files:
<...files....> Automatic merge failed; fix conflicts and then commit the result.`